web/src/list_widget.js

```diff
@@ -70,6 +71,17 @@ export function get_filtered_items(value, list, opts) {
     return list.filter((item) => predicate(item));
 }
 
+export function get_filtered_roles(value, list, opts) {
+    if (value === 0) {
+        return list;
+    }
+
+    if (opts.filter.role_filterer) {
+        return opts.filter.role_filterer(list, value);
+    }
+    return list;
+}
+
```

Comment:
@sahil839 , you mentioned above this comment:

> Though, I am not sure about the approach here. I think we should only have a single filter function and it should consider all the filters available for the table - in this case both search input and the role dropdown rather than passing filter and then role_filter separately. 

so do you want me to combine this and the above `get_filtered_items` function together or do you have something else in mind

Reply:
Yeah. Basically, a single function `get_filtered_items` should return the filtered list considering the search input and any other filters option present, which is role in this case.

Reply:
@sahil839 , just to confirm once before starting the work on this, so `settings_users.js` initiates the `list_widget` and sends 2 different filters option, `filterer` and `role_filterer`, and then we have two different filter functions as above, so do you want me to change the whole code path like merging both the search and filter option together from the starting, or just merge the above two functions so that we call only single function for the filter and leave everything same.